src/engine/controls/loopingcontrol.cpp

```diff
@@ -1400,13 +1400,17 @@ void LoopingControl::slotBeatLoop(double beats, bool keepStartPoint, bool enable
         return;
     }
 
-    // If resizing an inactive loop by changing beatloop_size,
-    // do not seek to the adjusted loop.
-    newloopInfo.seekMode = (keepStartPoint && (enable || m_bLoopingEnabled))
-            ? LoopSeekMode::Changed
-            : LoopSeekMode::MovedOut;
+    // Only update loopinfo if the endpoints have changed, otherwise we may clobber the existing
+    // LoopSeekMode selection.
+    if (!currentLoopMatchesBeatloopSize()) {
+        // If resizing an inactive loop by changing beatloop_size,
+        // do not seek to the adjusted loop.
+        newloopInfo.seekMode = (keepStartPoint && (enable || m_bLoopingEnabled))
+                ? LoopSeekMode::Changed
+                : LoopSeekMode::MovedOut;
 
-    m_loopInfo.setValue(newloopInfo);
+        m_loopInfo.setValue(newloopInfo);
```

Comment:
Is this move into the block correct? 
This way the CO and the m_loopInfo are out of sync.

Reply:
that might cause the asserts I'm seeing.  I'll tweak this